lib/platform/gitlab/index.md

```diff
@@ -3,3 +3,13 @@
 ## Features awaiting implementation
 
 - The `automergeStrategy` configuration option has not been implemented for this platform, and all values behave as if the value `auto` was used. Renovate will accept the Merge Request without further configuration, and respect the strategy defined in the Merge Request, and this cannot be overridden yet
+
+## Server version dependent features
+
+We use the GitLab [version API](https://docs.gitlab.com/ee/api/version.html) to fetch the server version.
+You can use the experimental feature flag [`RENOVATE_X_SERVER_VERSION`](https://docs.renovatebot.com/self-hosted-experimental/#renovate_x_server_version) to set a specific server version.
+By setting the server version yourself, you save a API call that fetches the server version.
```

Comment:
🤔  This needs to be set if you like to use `CI_JOB_TOKEN` because it doesn't have access right to the version api.

Reply:
It's best to use `CI_SERVER_VERSION` to set this in gitlab pipelines

Reply:
> 🤔 This needs to be set if you like to use `CI_JOB_TOKEN` because it doesn't have access right to the version api.

I'm not sure what this means. It sounds like some kind of "bootstrap problem"?

---

I noticed GitLab has a `CI_SERVER_VERSION_MAJOR` predefined variable? Would supporting that be easier than "pinning" yourself to a specific GitLab server version?

Reply:
We need the server major and minor version to check for specific features and i don't like to pass multiple variables.
This shoul dbe reusable for other platforms later if required.

if you like to use `CI_JOB_TOKEN` for authentication against gitlab, you don't have access to `api/v4/user` and `api/v4/version`. So thats why i like to override it, so we don't need to query the api. The user endpoint is already skiped, when you manually define `gitAuthor`, so only the `version` api request is left. I don't like to add an official option for it yet, maybe later, if this is really required